website/docs/components/code-block/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,100 @@
+## Usage
+
+### When to use
+
+- When displaying code examples and longer snippets of code that benefit from syntax highlighting.
+
+### When not to use
+
+- As a full-blown code editor
+- When displaying keys, tokens, or other generated values, use a [CopySnippet](/components/copy/snippet) instead.
+
+## Inline
+
+The `isInline` property increases the portability of the CodeBlock to ensure that it can be used in different contexts. For example, a common use case of the CodeBlock is in a "standalone" context, which can be part of a form, multi-step process, and is generally a part of the normal layout flow.
+
+![Code Block with rounded corners in a standalone context](/assets/components/code-block/code-block-rounded-standalone.png)
+
+Sometimes it may be necessary to use the CodeBlock in a more dense layout or nested within another component. In this circumstance, setting `isInline` to true ensures that the CodeBlock fits alongside other elements, in a split view, or as part of a larger layout mechanism.
+
+![Code Block in a block context](/assets/components/code-block/code-block-block-level.png)
+
+## Header
+
+Use a `title` or `description` in the Header to provide additional information, instructions, or to label a CodeBlock. Both of these properties are optional, but including them can help to provide additional context about a specific block of code.
+
+![Example metadata in the Code Block](/assets/components/code-block/code-block-metadata.png)
+
+### When not to use a Header
+
+There can be an overlap between content that you may choose to include in the Header as a `title` or `description`, and content that is part of the normal layout flow in a headline or paragraph. If it is necessary to elevate this content in the hierarchy of the page, we recommend including it in the normal layout flow, rather than as a `title` or `description` within the Code Block.
+
+![An example showcasing the Code Block paired with content in the nautral flow](/assets/components/code-block/code-block-dont-use-metadata.png)
+
+## CopyButton
+
+Use a CopyButton within the CodeBlock to make copying the snippet a single action. More details can be found in the [CopyButton guidelines](/components/copy/button).
+
+![An example of the Copy Button within the Code Block](/assets/components/code-block/code-block-copy-button.png)
+
+### Line selection
+
+If a user needs to copy only a portion of the CodeBlock, the relevant portion can be selected with a cursor and copied via the keyboard or mouse.
```

Comment:
There is a token (`.selected` I believe) in Prism that targets cursor selection, the approach that Kristin and I took was to let the browser defaults persist here, though that differs depending on the browser. I'm not sure about the accessibility requires around selection, @MelSumner can you chime in on this?

Reply:
[I mocked up a couple of different options here](https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/branch/5gAU4mfvx8DvfXKb4QpjoB/HDS-Product---Components?type=design&node-id=48316%3A54433&mode=design&t=TYzWMOx85f80K3Tb-1) for exploration

Reply:
@didoo do you have any opinion on this, is this something to include in the component?

cc @alex-ju for visibility as it will require a change in https://github.com/hashicorp/design-system/pull/1687

Reply:
so far we've not altered the selection style on either non-editable content (e.g. Alert or SideNav) or editable content (e.g. Textarea), but we can introduce it here if we want to. it is technically possible to change both the background and the foreground. 

If we change the background to white we risk making the contrast even worse; below there's a default example.

<img width="746" alt="Screenshot 2023-10-25 at 14 56 06" src="https://github.com/hashicorp/design-system/assets/788096/044bcda2-04c2-4845-96bc-8c3b2c42be92">

If we change both the foreground (unitarily) and the background we'll lose the highlight.

Reply:
@alex-ju and I discussed this a bit, if we are going to override the browser default, I would opt for either:

1. Inverting the foreground and background, so the text is dark and the background is light, [like in this example](https://www.figma.com/file/1TLwCzn7Bjkhw6OmCTqbPN/%F0%9F%92%80-Graveyard---FY24-Q3%2FQ4?type=design&node-id=52%3A336798&mode=design&t=2NW3TKQr1qb8Ouai-1)
2. Inverted the foreground so the text is dark, but the background is `green` syntax color [like this example](https://www.figma.com/file/1TLwCzn7Bjkhw6OmCTqbPN/%F0%9F%92%80-Graveyard---FY24-Q3%2FQ4?type=design&node-id=52%3A336750&mode=design&t=2NW3TKQr1qb8Ouai-1). Since there's generally a lot of blue in our product interfaces, the green seems like a better contrast to the similar "action" or "interactive" colors.

I'm not overly opinionated and don't generally mind the browser default, but we are going to adjust it, I would opt for one of the above options.